backend/app/models/chat.py

```diff
@@ -2,45 +2,44 @@
 from datetime import datetime
 from sqlalchemy import Column, DateTime, text, event
 from sqlmodel import Field, Relationship, SQLModel
+from typing import TYPE_CHECKING
 
+if TYPE_CHECKING:
+  from app.models.course import Course
 
-from app.models.course import Course
-
-
-# shared properties
 class ChatBase(SQLModel):
-  message: str | None =  Field(default=None, max_length=1024)
-  course_id: str = Field(min_length=3, max_length=255)
+  message: str | None = Field(default=None, max_length=1024)
   is_system: bool
   created_at: datetime = Field(
     sa_column=Column(DateTime(timezone=True), server_default=text("now()"), nullable=False)
   )
   updated_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)
 
 # Properties to receive on chat creation
-class ChatCreate(ChatBase):
-  message: str
-  course_id: str
+class ChatCreate(SQLModel):  # Don't inherit from ChatBase to avoid conflicts
+  message: str = Field(max_length=1024)
+  course_id: uuid.UUID
   is_system: bool
 
 # Properties to receive on chat update
-class ChatUpdate(ChatBase):
-  id: str
-  message: str
+class ChatUpdate(SQLModel):
+  message: str | None = Field(default=None, max_length=1024)
 
-
-# Database model, database table inferred from class name
 class Chat(ChatBase, table=True):
-  __tablename__ = "chat"
-
   id: uuid.UUID = Field(default_factory=uuid.uuid4, primary_key=True)
-  message: str = Field(nullable=False, max_length=1024)
-  is_system: bool = Field(nullable=False)
-  course_id: uuid.UUID = Field(foreign_key="course.id", nullable=False, ondelete="CASCADE")
+  course_id: uuid.UUID = Field(foreign_key="course.id", ondelete="CASCADE")
+  course: "Course" = Relationship(back_populates="chats")
 
-  course: Course | None = Relationship(back_populates="course")
+# Public model for API responses
+class ChatPublic(SQLModel):
```

Comment:
Move the Pydantic schema to the `/public/schema.py`, and this will resolve circular dependency that keeps coming up